[question] How to build for Mac Catalyst?

<!--
  Please don't forget to update the issue title.
  Include all applicable information to help us reproduce your problem.

  To help us debug your issue please explain:
-->

1.33 with catalyst support is out and I tried it right away. Maybe with a bit naive first try. :-)

### Environment Details (include every applicable attribute)
  * Operating System+version: Catalina
  * Compiler+version: Apple clang version 12.0.0 (clang-1200.0.32.28)
  * Conan version: 1.33
  * Python version:  3.7.4

### Steps to reproduce (Include if Applicable)
conan install xmlsec/1.2.31@ --build --profile ./conan/profiles/mac_catalyst

### Error
from config.log

>configure:4198: $? = 0
configure:4187: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -V >&5
clang: error: argument to '-V' is missing (expected 1 value)
clang: error: no input files
configure:4198: $? = 1
configure:4187: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -qversion >&5
**clang: error: unknown argument '-qversion'; did you mean '--version'?**
clang: error: no input files
configure:4198: $? = 1
configure:4218: checking whether the C compiler works
configure:4240: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -target=x86_64-apple-ios-macabi -O3 --sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fPIC -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -DNDEBUG -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -target=x86_64-apple-ios-macabi --sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 conftest.c  >&5
**clang: error: unknown argument '-target=x86_64-apple-ios-macabi'; did you mean '--target=x86_64-apple-ios-macabi'?
clang: error: unknown argument '-target=x86_64-apple-ios-macabi'; did you mean '--target=x86_64-apple-ios-macabi'?**
configure:4244: $? = 1

______
???
configure:4240: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang 
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64
-target=x86_64-apple-ios-macabi
-O3
--sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-fPIC 
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64 -DNDEBUG
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk 
-arch x86_64
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64 -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64
-target=x86_64-apple-ios-macabi
--sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk 
-arch x86_64 conftest.c  >&5

### Profile
>[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=14
compiler.libcxx=libc++
compiler.version=12.0
os=Macos
os.subsystem=Catalyst
os_build=Macos
[options]
[build_requires]
*: darwin-toolchain/1.0.8@theodelrieu/stable
[env]

### Logs (Executed commands with output) (Include/Attach if Applicable)

<!--
  Your log content should be related to the bug description, it can be:
  - Conan command output
  - Server output (Artifactory, conan_server)
-->

```
conan install xmlsec/1.2.31@ --build --profile ./conan/profiles/mac_catalyst
Configuration:
[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=14
compiler.libcxx=libc++
compiler.version=12.0
os=Macos
os.subsystem=Catalyst
os_build=Macos
[options]
[build_requires]
*: darwin-toolchain/1.0.8@theodelrieu/stable
[env]

libiconv/1.16: Forced build from source
openssl/1.1.1h: Forced build from source
zlib/1.2.11: Forced build from source
libxml2/2.9.10: Forced build from source
xmlsec/1.2.31: Forced build from source
darwin-toolchain/1.0.8@theodelrieu/stable: Forced build from source
m4/1.4.18: Forced build from source
pkgconf/1.7.3: Forced build from source
autoconf/2.69: Forced build from source
automake/1.16.3: Forced build from source
libtool/2.4.6: Forced build from source
ninja/1.10.2: Forced build from source
meson/0.55.1: Forced build from source
Installing package: xmlsec/1.2.31
Requirements
    libiconv/1.16 from 'conan-center' - Cache
    libxml2/2.9.10 from 'conan-center' - Cache
    openssl/1.1.1h from 'conan-center' - Cache
    xmlsec/1.2.31 from 'conan-center' - Cache
    zlib/1.2.11 from 'conan-center' - Cache
Packages
    libiconv/1.16:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build
    libxml2/2.9.10:a2fcfc5995ddeefb1ff988dbc9aa90b909d43ed5 - Build
    openssl/1.1.1h:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build
    xmlsec/1.2.31:6c9b5886a0769be2cb726e1b2e87b7bc5c550492 - Build
    zlib/1.2.11:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build
Build requirements
    autoconf/2.69 from 'conan-center' - Cache
    automake/1.16.3 from 'conan-center' - Cache
    darwin-toolchain/1.0.8@theodelrieu/stable from 'conan-center' - Cache
    libtool/2.4.6 from 'conan-center' - Cache
    m4/1.4.18 from 'conan-center' - Cache
    meson/0.55.1 from 'conan-center' - Cache
    ninja/1.10.2 from 'conan-center' - Cache
    pkgconf/1.7.3 from 'conan-center' - Cache
Build requirements packages
    autoconf/2.69:923c5310a94256af03d94c969eebbbadac513c42 - Build
    automake/1.16.3:8198ae767c7bf455a38775936906b3e982c035ec - Build
    darwin-toolchain/1.0.8@theodelrieu/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Build
    libtool/2.4.6:072e844e71af4f1016fea59345350a58177d9979 - Build
    m4/1.4.18:3bc8a0eb26ab34ed4958480d81dfd67c24f74ed4 - Build
    meson/0.55.1:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Build
    ninja/1.10.2:3f3351f8c9c8848e68a09e5f612a66c58eb79473 - Build
    pkgconf/1.7.3:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build

Installing (downloading, building) binaries...
darwin-toolchain/1.0.8@theodelrieu/stable: Copying sources to build folder
darwin-toolchain/1.0.8@theodelrieu/stable: Building your package in /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
darwin-toolchain/1.0.8@theodelrieu/stable: Generator txt created conanbuildinfo.txt
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] 'fPIC' option not found
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
darwin-toolchain/1.0.8@theodelrieu/stable: Calling build()
darwin-toolchain/1.0.8@theodelrieu/stable: WARN: This conanfile has no build step
darwin-toolchain/1.0.8@theodelrieu/stable: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' built
darwin-toolchain/1.0.8@theodelrieu/stable: Build folder /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
darwin-toolchain/1.0.8@theodelrieu/stable: Generated conaninfo.txt
darwin-toolchain/1.0.8@theodelrieu/stable: Generated conanbuildinfo.txt
darwin-toolchain/1.0.8@theodelrieu/stable: Generating the package
darwin-toolchain/1.0.8@theodelrieu/stable: Package folder /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
darwin-toolchain/1.0.8@theodelrieu/stable: Calling package()
[HOOK - conan-center.py] post_package(): ERROR: [PACKAGE LICENSE (KB-H012)] No 'licenses' folder found in package: /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9  (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H012)
[HOOK - conan-center.py] post_package(): ERROR: [DEFAULT PACKAGE LAYOUT (KB-H013)] Unknown file 'darwin-toolchain.cmake' in the package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H013)
[HOOK - conan-center.py] post_package(): [DEFAULT PACKAGE LAYOUT (KB-H013)] If you are trying to package a tool put all the contents under the 'bin' folder
[HOOK - conan-center.py] post_package(): ERROR: [MATCHING CONFIGURATION (KB-H014)] Empty package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H014)
[HOOK - conan-center.py] post_package(): ERROR: [MATCHING CONFIGURATION (KB-H014)] Packaged artifacts does not match the settings used: os=Macos, compiler=apple-clang (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H014)
[HOOK - conan-center.py] post_package(): [SHARED ARTIFACTS (KB-H015)] OK
[HOOK - conan-center.py] post_package(): [PC-FILES (KB-H020)] OK
[HOOK - conan-center.py] post_package(): [CMAKE-MODULES-CONFIG-FILES (KB-H016)] OK
[HOOK - conan-center.py] post_package(): [PDB FILES NOT ALLOWED (KB-H017)] OK
[HOOK - conan-center.py] post_package(): [LIBTOOL FILES PRESENCE (KB-H018)] OK
[HOOK - conan-center.py] post_package(): [MS RUNTIME FILES (KB-H021)] OK
darwin-toolchain/1.0.8@theodelrieu/stable package(): Packaged 1 '.cmake' file: darwin-toolchain.cmake
darwin-toolchain/1.0.8@theodelrieu/stable: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' created
darwin-toolchain/1.0.8@theodelrieu/stable: Created package revision c093366e06ad120604635abd2f99a5d2
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
libiconv/1.16: Applying build-requirement: darwin-toolchain/1.0.8@theodelrieu/stable
libiconv/1.16: WARN: Build folder is dirty, removing it: /Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701
libiconv/1.16: Copying sources to build folder
libiconv/1.16: Building your package in /Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701
libiconv/1.16: Generator txt created conanbuildinfo.txt
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK. 'fPIC' option found and apparently well managed
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
libiconv/1.16: Calling build()
libiconv/1.16: Calling:
 > ./configure '--enable-static' '--disable-shared' '--prefix=/Users/dmn/.conan/data/libiconv/1.16/_/_/package/9d3239a97b703c82f38d9e91676ca971d7eb7701' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' '--datarootdir=${prefix}/share'
checking for a BSD-compatible install... /usr/local/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/local/bin/gmkdir -p
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make sets $(MAKE)... (cached) yes
checking for gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
checking whether the C compiler works... no
configure: error: in `/Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701/source_subfolder':
configure: error: C compiler cannot create executables
See `config.log' for more details
libiconv/1.16:
libiconv/1.16: ERROR: Package '9d3239a97b703c82f38d9e91676ca971d7eb7701' build failed
libiconv/1.16: WARN: Build folder /Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701
ERROR: libiconv/1.16: Error in build() method, line 127
	autotools = self._configure_autotools()
while calling '_configure_autotools', line 117
	self._autotools.configure(args=configure_args, host=host, build=build)
	ConanException: Error 77 while executing ./configure '--enable-static' '--disable-shared' '--prefix=/Users/dmn/.conan/data/libiconv/1.16/_/_/package/9d3239a97b703c82f38d9e91676ca971d7eb7701' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' '--datarootdir=${prefix}/share'
```

--- a/conans/__init__.py
+++ b/conans/__init__.py
@@ -88,4 +88,4 @@ OAUTH_TOKEN = "oauth_token"
 SERVER_CAPABILITIES = [COMPLEX_SEARCH_CAPABILITY, REVISIONS]  # Server is always with revisions
 DEFAULT_REVISION_V1 = "0"
 
-__version__ = '1.33.0'
+__version__ = '1.33.1'

--- a/conans/client/build/autotools_environment.py
+++ b/conans/client/build/autotools_environment.py
@@ -353,7 +353,8 @@ class AutoToolsBuildEnvironment(object):
                 concat += " " + os.environ.get("CFLAGS", None)
             if os.environ.get("CXXFLAGS", None):
                 concat += " " + os.environ.get("CXXFLAGS", None)
-            if self._os_version and "-version-min" not in concat and "-target" not in concat:
+            if (self._os_version and "-version-min" not in concat and "-target" not in concat) or \
+                    self._os_subsystem:
                 tmp_compilation_flags.append(tools.apple_deployment_target_flag(self._os,
                                                                                 self._os_version,
                                                                                 self._os_sdk,

--- a/conans/client/build/compiler_flags.py
+++ b/conans/client/build/compiler_flags.py
@@ -47,8 +47,8 @@ def architecture_flag(settings):
         return ""
 
     if str(compiler) in ['gcc', 'apple-clang', 'clang', 'sun-cc']:
-        if str(the_os) == 'Macos' and str(subsystem) == 'Catalyst' and str(arch) == 'x86_64':
-            return '-target=x86_64-apple-ios-macabi'
+        if str(the_os) == 'Macos' and str(subsystem) == 'catalyst' and str(arch) == 'x86_64':
+            return '--target=x86_64-apple-ios-macabi'
         elif str(arch) in ['x86_64', 'sparcv9', 's390x']:
             return '-m64'
         elif str(arch) in ['x86', 'sparc']:

--- a/conans/client/conf/__init__.py
+++ b/conans/client/conf/__init__.py
@@ -36,9 +36,9 @@ _t_default_settings_yml = Template(textwrap.dedent("""
             version: ["5.0", "6.0", "7.0", "8.0"]
         Linux:
         Macos:
-            version: [None, "10.6", "10.7", "10.8", "10.9", "10.10", "10.11", "10.12", "10.13", "10.14", "10.15", "11.0"]
+            version: [None, "10.6", "10.7", "10.8", "10.9", "10.10", "10.11", "10.12", "10.13", "10.14", "10.15", "11.0", "13.0"]
             sdk: [None, "macosx"]
-            subsystem: [None, "Catalyst"]
+            subsystem: [None, catalyst]
         Android:
             api_level: ANY
         iOS:

--- a/conans/client/migrations_settings.py
+++ b/conans/client/migrations_settings.py
@@ -1936,3 +1936,123 @@ build_type: [None, Debug, Release, RelWithDebInfo, MinSizeRel]
 
 cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]  # Deprecated, use compiler.cppstd
 """
+
+settings_1_33_1 = """
+# Only for cross building, 'os_build/arch_build' is the system that runs Conan
+os_build: [Windows, WindowsStore, Linux, Macos, FreeBSD, SunOS, AIX]
+arch_build: [x86, x86_64, ppc32be, ppc32, ppc64le, ppc64, armv5el, armv5hf, armv6, armv7, armv7hf, armv7s, armv7k, armv8, armv8_32, armv8.3, sparc, sparcv9, mips, mips64, avr, s390, s390x, sh4le, e2k-v2, e2k-v3, e2k-v4, e2k-v5, e2k-v6, e2k-v7]
+
+# Only for building cross compilation tools, 'os_target/arch_target' is the system for
+# which the tools generate code
+os_target: [Windows, Linux, Macos, Android, iOS, watchOS, tvOS, FreeBSD, SunOS, AIX, Arduino, Neutrino]
+arch_target: [x86, x86_64, ppc32be, ppc32, ppc64le, ppc64, armv5el, armv5hf, armv6, armv7, armv7hf, armv7s, armv7k, armv8, armv8_32, armv8.3, sparc, sparcv9, mips, mips64, avr, s390, s390x, asm.js, wasm, sh4le, e2k-v2, e2k-v3, e2k-v4, e2k-v5, e2k-v6, e2k-v7]
+
+# Rest of the settings are "host" settings:
+# - For native building/cross building: Where the library/program will run.
+# - For building cross compilation tools: Where the cross compiler will run.
+os:
+    Windows:
+        subsystem: [None, cygwin, msys, msys2, wsl]
+    WindowsStore:
+        version: ["8.1", "10.0"]
+    WindowsCE:
+        platform: ANY
+        version: ["5.0", "6.0", "7.0", "8.0"]
+    Linux:
+    Macos:
+        version: [None, "10.6", "10.7", "10.8", "10.9", "10.10", "10.11", "10.12", "10.13", "10.14", "10.15", "11.0", "13.0"]
+        sdk: [None, "macosx"]
+        subsystem: [None, catalyst]
+    Android:
+        api_level: ANY
+    iOS:
+        version: ["7.0", "7.1", "8.0", "8.1", "8.2", "8.3", "9.0", "9.1", "9.2", "9.3", "10.0", "10.1", "10.2", "10.3", "11.0", "11.1", "11.2", "11.3", "11.4", "12.0", "12.1", "12.2", "12.3", "12.4", "13.0", "13.1", "13.2", "13.3", "13.4", "13.5", "13.6"]
+        sdk: [None, "iphoneos", "iphonesimulator"]
+    watchOS:
+        version: ["4.0", "4.1", "4.2", "4.3", "5.0", "5.1", "5.2", "5.3", "6.0", "6.1"]
+        sdk: [None, "watchos", "watchsimulator"]
+    tvOS:
+        version: ["11.0", "11.1", "11.2", "11.3", "11.4", "12.0", "12.1", "12.2", "12.3", "12.4", "13.0"]
+        sdk: [None, "appletvos", "appletvsimulator"]
+    FreeBSD:
+    SunOS:
+    AIX:
+    Arduino:
+        board: ANY
+    Emscripten:
+    Neutrino:
+        version: ["6.4", "6.5", "6.6", "7.0", "7.1"]
+arch: [x86, x86_64, ppc32be, ppc32, ppc64le, ppc64, armv4, armv4i, armv5el, armv5hf, armv6, armv7, armv7hf, armv7s, armv7k, armv8, armv8_32, armv8.3, sparc, sparcv9, mips, mips64, avr, s390, s390x, asm.js, wasm, sh4le, e2k-v2, e2k-v3, e2k-v4, e2k-v5, e2k-v6, e2k-v7]
+compiler:
+    sun-cc:
+        version: ["5.10", "5.11", "5.12", "5.13", "5.14", "5.15"]
+        threads: [None, posix]
+        libcxx: [libCstd, libstdcxx, libstlport, libstdc++]
+    gcc: &gcc
+        version: ["4.1", "4.4", "4.5", "4.6", "4.7", "4.8", "4.9",
+                  "5", "5.1", "5.2", "5.3", "5.4", "5.5",
+                  "6", "6.1", "6.2", "6.3", "6.4", "6.5",
+                  "7", "7.1", "7.2", "7.3", "7.4", "7.5",
+                  "8", "8.1", "8.2", "8.3", "8.4",
+                  "9", "9.1", "9.2", "9.3",
+                  "10", "10.1"]
+        libcxx: [libstdc++, libstdc++11]
+        threads: [None, posix, win32] #  Windows MinGW
+        exception: [None, dwarf2, sjlj, seh] # Windows MinGW
+        cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]
+    Visual Studio: &visual_studio
+        runtime: [MD, MT, MTd, MDd]
+        version: ["8", "9", "10", "11", "12", "14", "15", "16"]
+        toolset: [None, v90, v100, v110, v110_xp, v120, v120_xp,
+                  v140, v140_xp, v140_clang_c2, LLVM-vs2012, LLVM-vs2012_xp,
+                  LLVM-vs2013, LLVM-vs2013_xp, LLVM-vs2014, LLVM-vs2014_xp,
+                  LLVM-vs2017, LLVM-vs2017_xp, v141, v141_xp, v141_clang_c2, v142,
+                  llvm, ClangCL]
+        cppstd: [None, 14, 17, 20]
+    msvc:
+        version: ["19.0",
+                  "19.1", "19.10", "19.11", "19.12", "19.13", "19.14", "19.15", "19.16",
+                  "19.2", "19.20", "19.21", "19.22", "19.23", "19.24", "19.25", "19.26", "19.27", "19.28"]
+        runtime: [static, dynamic]
+        runtime_type: [Debug, Release]
+        cppstd: [14, 17, 20]
+    clang:
+        version: ["3.3", "3.4", "3.5", "3.6", "3.7", "3.8", "3.9", "4.0",
+                  "5.0", "6.0", "7.0", "7.1",
+                  "8", "9", "10", "11"]
+        libcxx: [None, libstdc++, libstdc++11, libc++, c++_shared, c++_static]
+        cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]
+        runtime: [None, MD, MT, MTd, MDd]
+    apple-clang: &apple_clang
+        version: ["5.0", "5.1", "6.0", "6.1", "7.0", "7.3", "8.0", "8.1", "9.0", "9.1", "10.0", "11.0", "12.0"]
+        libcxx: [libstdc++, libc++]
+        cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]
+    intel:
+        version: ["11", "12", "13", "14", "15", "16", "17", "18", "19", "19.1"]
+        base:
+            gcc:
+                <<: *gcc
+                threads: [None]
+                exception: [None]
+            Visual Studio:
+                <<: *visual_studio
+            apple-clang:
+                <<: *apple_clang
+    qcc:
+        version: ["4.4", "5.4", "8.3"]
+        libcxx: [cxx, gpp, cpp, cpp-ne, accp, acpp-ne, ecpp, ecpp-ne]
+        cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17]
+    mcst-lcc:
+        version: ["1.19", "1.20", "1.21", "1.22", "1.23", "1.24", "1.25"]
+        base:
+            gcc:
+                <<: *gcc
+                threads: [None]
+                exceptions: [None]
+
+build_type: [None, Debug, Release, RelWithDebInfo, MinSizeRel]
+
+
+cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]  # Deprecated, use compiler.cppstd
+"""
+

--- a/conans/client/tools/apple.py
+++ b/conans/client/tools/apple.py
@@ -65,7 +65,7 @@ def apple_deployment_target_flag(os_, os_version, os_sdk=None, os_subsystem=None
             'watchsimulator': '-mwatchos-simulator-version-min',
             'appletvos': '-mtvos-version-min',
             'appletvsimulator': '-mtvos-simulator-version-min'}.get(str(os_sdk))
-    if os_subsystem == 'Catalyst':
+    if os_subsystem == 'catalyst':
         # especial case, despite Catalyst is macOS, it requires an iOS version argument
         flag = '-mios-version-min'
     if not flag:

--- a/conans/test/functional/build_helpers/autotools_apple_test.py
+++ b/conans/test/functional/build_helpers/autotools_apple_test.py
@@ -12,6 +12,41 @@ from conans.test.utils.tools import TestClient
 
 @unittest.skipUnless(platform.system() == "Darwin", "requires Xcode")
 class AutoToolsAppleTest(unittest.TestCase):
+    makefile = textwrap.dedent("""
+        .PHONY: all
+        all: libhello.a app
+
+        app: main.o libhello.a
+        	$(CXX) $(CFLAGS) -o app main.o -lhello -L.
+
+        libhello.a: hello.o
+        	$(AR) rcs libhello.a hello.o
+
+        main.o: main.cpp
+        	$(CXX) $(CFLAGS) -c -o main.o main.cpp
+
+        hello.o: hello.cpp
+        	$(CXX) $(CFLAGS) -c -o hello.o hello.cpp
+        """)
+
+    conanfile_py = textwrap.dedent("""
+        from conans import ConanFile, tools, AutoToolsBuildEnvironment
+
+
+        class App(ConanFile):
+            settings = "os", "arch", "compiler", "build_type"
+            options = {"shared": [True, False], "fPIC": [True, False]}
+            default_options = {"shared": False, "fPIC": True}
+
+            def config_options(self):
+                if self.settings.os == "Windows":
+                    del self.options.fPIC
+
+            def build(self):
+                env_build = AutoToolsBuildEnvironment(self)
+                env_build.make()
+        """)
+
     @parameterized.expand([("x86_64", "Macos", "10.14"),
                            ("armv8", "iOS", "10.0"),
                            ("armv7", "iOS", "10.0"),
@@ -23,23 +58,6 @@ class AutoToolsAppleTest(unittest.TestCase):
         self.os = os_
         self.os_version = os_version
 
-        makefile = textwrap.dedent("""
-            .PHONY: all
-            all: libhello.a app
-
-            app: main.o libhello.a
-            	$(CXX) $(CFLAGS) -o app main.o -lhello -L.
-
-            libhello.a: hello.o
-            	$(AR) rcs libhello.a hello.o
-
-            main.o: main.cpp
-            	$(CXX) $(CFLAGS) -c -o main.o main.cpp
-
-            hello.o: hello.cpp
-            	$(CXX) $(CFLAGS) -c -o hello.o hello.cpp
-            """)
-
         profile = textwrap.dedent("""
             include(default)
             [settings]
@@ -48,34 +66,16 @@ class AutoToolsAppleTest(unittest.TestCase):
             arch = {arch}
             """).format(os=self.os, arch=self.arch, os_version=self.os_version)
 
-        conanfile_py = textwrap.dedent("""
-            from conans import ConanFile, tools, AutoToolsBuildEnvironment
-
-
-            class App(ConanFile):
-                settings = "os", "arch", "compiler", "build_type"
-                options = {"shared": [True, False], "fPIC": [True, False]}
-                default_options = {"shared": False, "fPIC": True}
-
-                def config_options(self):
-                    if self.settings.os == "Windows":
-                        del self.options.fPIC
-
-                def build(self):
-                    env_build = AutoToolsBuildEnvironment(self)
-                    env_build.make()
-            """)
-
         self.t = TestClient()
         hello_h = gen_function_h(name="hello")
         hello_cpp = gen_function_cpp(name="hello")
         main_cpp = gen_function_cpp(name="main", includes=["hello"], calls=["hello"])
 
-        self.t.save({"Makefile": makefile,
+        self.t.save({"Makefile": self.makefile,
                      "hello.h": hello_h,
                      "hello.cpp": hello_cpp,
                      "main.cpp": main_cpp,
-                     "conanfile.py": conanfile_py,
+                     "conanfile.py": self.conanfile_py,
                      "profile": profile})
 
         self.t.run("install . --profile:host=profile")
@@ -93,3 +93,56 @@ class AutoToolsAppleTest(unittest.TestCase):
 
         self.t.run_command('lipo -info "%s"' % app)
         self.assertIn("architecture: %s" % expected_arch, self.t.out)
+
+    def test_catalyst(self):
+        profile = textwrap.dedent("""
+            include(default)
+            [settings]
+            os = Macos
+            os.version = 13.0
+            os.sdk = macosx
+            os.subsystem = catalyst
+            arch = x86_64
+            """)
+
+        self.t = TestClient()
+        hello_h = gen_function_h(name="hello")
+        hello_cpp = gen_function_cpp(name="hello")
+        main_cpp = textwrap.dedent("""
+            #include "hello.h"
+            #include <TargetConditionals.h>
+            #include <iostream>
+
+            int main()
+            {
+            #if TARGET_OS_MACCATALYST
+                std::cout << "running catalyst " << __IPHONE_OS_VERSION_MIN_REQUIRED << std::endl;
+            #else
+                #error "not building for Apple Catalyst"
+            #endif
+            }
+            """)
+
+        self.t.save({"Makefile": self.makefile,
+                     "hello.h": hello_h,
+                     "hello.cpp": hello_cpp,
+                     "main.cpp": main_cpp,
+                     "conanfile.py": self.conanfile_py,
+                     "profile": profile})
+
+        self.t.run("install . --profile:host=profile")
+        self.t.run("build .")
+
+        libhello = os.path.join(self.t.current_folder, "libhello.a")
+        app = os.path.join(self.t.current_folder, "app")
+        self.assertTrue(os.path.isfile(libhello))
+        self.assertTrue(os.path.isfile(app))
+
+        self.t.run_command('lipo -info "%s"' % libhello)
+        self.assertIn("architecture: x86_64", self.t.out)
+
+        self.t.run_command('lipo -info "%s"' % app)
+        self.assertIn("architecture: x86_64", self.t.out)
+
+        self.t.run_command('"%s"' % app)
+        self.assertIn("running catalyst 130000", self.t.out)

--- a/conans/test/unittests/client/build/compiler_flags_test.py
+++ b/conans/test/unittests/client/build/compiler_flags_test.py
@@ -47,8 +47,8 @@ class CompilerFlagsTest(unittest.TestCase):
         settings = MockSettings({"compiler": "apple-clang",
                                  "arch": "x86_64",
                                  "os": "Macos",
-                                 "os.subsystem": "Catalyst"})
-        self.assertEqual(architecture_flag(settings), "-target=x86_64-apple-ios-macabi")
+                                 "os.subsystem": "catalyst"})
+        self.assertEqual(architecture_flag(settings), "--target=x86_64-apple-ios-macabi")
 
     @parameterized.expand([("gcc", "x86", "-m32"),
                            ("gcc", "x86_64", "-m64"),

--- a/conans/test/unittests/util/apple_test.py
+++ b/conans/test/unittests/util/apple_test.py
@@ -115,10 +115,10 @@ class AppleTest(unittest.TestCase):
         self.assertEqual(tools.apple_deployment_target_flag('tvOS', "10.1", 'appletvsimulator'),
                          '-mtvos-simulator-version-min=10.1')
 
-        self.assertEqual(tools.apple_deployment_target_flag("Macos", "10.1", None, "Catalyst"),
+        self.assertEqual(tools.apple_deployment_target_flag("Macos", "10.1", None, "catalyst"),
                          '-mios-version-min=10.1')
 
-        self.assertEqual(tools.apple_deployment_target_flag("Macos", "10.1", "macosx", "Catalyst"),
+        self.assertEqual(tools.apple_deployment_target_flag("Macos", "10.1", "macosx", "catalyst"),
                          '-mios-version-min=10.1')
 
         self.assertEqual('', tools.apple_deployment_target_flag('Solaris', "10.1"))

